website/static/js/filepage/index.js

```diff
@@ -161,41 +161,16 @@ var FileViewPage = {
         //This code was abstracted into a panel toggler at one point
         //it was removed and shoved here due to issues with mithrils caching and interacting
         //With other non-mithril components on the page
-        var panels;
-        if (ctrl.editor) {
-            panels = [ctrl.editor, ctrl.revisions];
-        } else {
-            panels = [ctrl.revisions];
-        }
+        ctrl.enableEditing();
 
-        var shown = panels.reduce(function(accu, panel) {
-            return accu + (panel.selected ? 1 : 0);
-        }, 0);
-
-        var panelsShown = shown + (ctrl.mfrIframeParent.is(':visible') ? 1 : 0);
+        var panelsShown = ((ctrl.editor && ctrl.editor.selected) ? 1 : 0) + (ctrl.mfrIframeParent.is(':visible') ? 1 : 0);
```

Comment:
The ternary operators being used to conditionally include panels, then counting the total is complex enough that I believe it would warrant at least a comment line. I'd probably break it up more like this, too:

```javascript
var panelsShown = (
    ((ctrl.editor && ctrl.editor.selected) ? 1 : 0)  // Editor panel is active
    + (ctrl.mfrIframeParent.is(':visible') ? 1 : 0)  // View panel is active
);
```

Alternatively, you could spilt it up further by implementing a counter:
```javascript
var panelsShown = 0;
if (ctrl.editor && ctrl.editor.selected) {
    // Editor panel is active
    panelsShown++;
}
if (ctrl.mfrIframeParent.is(':visible')) {
    // View panel is active
    panelsShown++;
}
```

Neither of these are necessarily better than the way it's done now, and the second approach is less performant - so it's up to you if you want to change it. I only know that there is it took me a moment to understand what was happening on this line.